lib/set-task.js

```diff
@@ -15,12 +15,26 @@ function set(name, fn){
 
   var meta = metadata.get(fn) || {};
   var nodes = [];
-  if(meta.tree){
+  if(meta.tree && meta.tree.nodes.length){
     nodes.push(meta.tree);
+  } else if (!meta.name) {
+    meta.name = fn.displayName || fn.name || '<anonymous>';
+    meta.tree = {
+      label: meta.name,
+      type: 'function',
+      nodes: []
+    };
+    metadata.set(fn, meta);
   }
 
   var task = this._registry.set(name, fn) || fn;
 
+  if (task === fn) {
+    task = fn.bind();
+    fn.tasks = fn.tasks || {};
+    fn.tasks[name] = task;
```

Comment:
I don't think mutating the passed function should be done.  The only mutation that is caused is due to the weakmap shim